### DISCUSSION TITLE
Math Formula Editor

### DISCUSSION THREAD

**Discussion by stefnotch** [mgreminger/EngineeringPaper.xyz#135] [General]
[EngineeringPaper](https://github.com/mgreminger/EngineeringPaper.xyz) currently uses the lovely MathQuill editor. I'm guessing MathQuill was chosen, since it's the most reliable open-source WYSIWYG math editor. The main downside of using MathQuill is the fact that matrices aren't supported, unless one uses a fork.

As the author of QuantumSheet, I'm currently writing my own math formula editor. The goal is to take everything that makes IDEs nice and put those features into a modern math formula editor. That means autocompletion, documentation popups, unambiguous input with the editor asking the user when something is underspecified, ....
https://github.com/stefnotch/mathml-editor
(Demo is woefully incomplete, the editor is a work in progress)

I figured it'd be worthwhile to reach out, and ask if you have any thoughts regarding what a truly good math formula editor should be like.

**Comment by mgreminger** [comment]
@stefnotch, interesting project, thanks for sharing. Yes, the lack of matrix support is one of the limitations of MathQuill. Matrix support is probably a ways off for EngineeringPaper.xyz, but it will likely require a different math editor to support it. I'm going to move this over to a discussion item since it probably fits better over there.
